core/src/Kokkos_Macros.hpp

```diff
@@ -607,6 +607,18 @@ static constexpr bool kokkos_omp_on_host() { return false; }
   #define KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH()
   #define KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_POP()
 #endif
+
+#if defined(__NVCOMPILER)
+#define KOKKOS_IMPL_DISABLE_UNREACHABLE_WARNINGS_PUSH() \
+  _Pragma("diag_suppress code_is_unreachable")          \
+  _Pragma("diag_suppress initialization_not_reachable")
```

Comment:
We actually need both?

Reply:
I couldn't find a catch all warning for unreachable code for nvc++. We need `initialization_not_reachable` because of 
```
TEST(TEST_CATEGORY, IncrTest_12a_ThreadScratch) {
#ifdef KOKKOS_ENABLE_OPENACC  // FIXME_OPENACC
  GTEST_SKIP() << "skipping since scratch memory is not yet implemented in the "
                  "OpenACC backend";
#endif

  ThreadScratch<TEST_EXECSPACE> test; // warning thrown here for initing test
  ...
}
```